external/avro/src/main/scala/org/apache/spark/sql/avro/package.scala

```diff
@@ -36,4 +40,27 @@ package object avro {
     @scala.annotation.varargs
     def avro(sources: String*): DataFrame = reader.format("avro").load(sources: _*)
   }
+
+  /**
+   * Converts a binary column of avro format into its corresponding catalyst value. The specified
+   * schema must match the read data, otherwise the behavior is undefined: it may fail or return
+   * arbitrary result.
+   *
+   * @param data the binary column.
+   * @param avroType the avro type.
+   */
+  @Experimental
+  def from_avro(data: Column, avroType: Schema): Column = {
```

Comment:
Why don't we just expose String version (json) in case we expose it in SQL syntax? I am sure we won't want to have many variants in the future as well.

Reply:
`Column` can support arbitrary expressions. Personally I don't like these string-based overloads...

Since it's in an external module, I think we can only add python/R/SQL version when we have a persistent UDF API.

Reply:
Yup, I was just worried if we happened to have many overloaded version of this again like json function family. We happened to have many overloaded versions of it, which seems an actual concern (just roughly given my few interaction with other guys like Reynold).

Reply:
ah sorry i thought you are talking about the `data` parameter.

Yes, for `avroType` parameter, we should have a string version

Reply:
I am actually +1 with @HyukjinKwon . 
To make everyone happy, I will add another override API, but I don't think we should have more.